src/Microsoft.DotNet.Build.Tasks.Packaging/src/GetApplicableAssetsFromPackages.cs

```diff
@@ -73,32 +73,60 @@ public override bool Execute()
+            NuGetFramework[] compileFxs = this.TargetMonikers.Select(fx => NuGetFramework.Parse(fx)).ToArray();
+            NuGetFramework[] runtimeFxs = compileFxs;
 
-            if (!String.IsNullOrEmpty(RuntimeTargetMoniker))
+            if (this.RuntimeTargetMonikers != null && this.TargetMonikers.Length > 0)
             {
-                runtimeFx = NuGetFramework.Parse(RuntimeTargetMoniker);
+                runtimeFxs = this.RuntimeTargetMonikers.Select(fx => NuGetFramework.Parse(fx)).ToArray();
             }
 
-            string targetString = String.IsNullOrEmpty(TargetRuntime) ? fx.ToString() : $"{fx}/{TargetRuntime}";
-
             LoadFiles();
 
-            CompileAssets = _resolver.ResolveCompileAssets(fx)
-                                .Where(ca => !NuGetAssetResolver.IsPlaceholder(ca))
-                                .Select(ca => PackageItemAsResolvedAsset(_targetPathToPackageItem[ca]))
-                                .ToArray();
-            RuntimeAssets = _resolver.ResolveRuntimeAssets(runtimeFx, TargetRuntime)
-                                .Where(ra => !NuGetAssetResolver.IsPlaceholder(ra))
-                                .Select(ra => PackageItemAsResolvedAsset(_targetPathToPackageItem[ra]))
-                                .ToArray();
+            // find the best framework
+            foreach (var compileFx in compileFxs)
+            {
+                CompileAssets = _resolver.ResolveCompileAssets(compileFx)
+                                         .Where(ca => !NuGetAssetResolver.IsPlaceholder(ca))
+                                         .Select(ca => PackageItemAsResolvedAsset(_targetPathToPackageItem[ca]))
+                                         .ToArray();
+
+                if (CompileAssets.Any())
+                {
+                    Log.LogMessage($"Resolved compile assets from {compileFx.ToString()}: {String.Join(";", CompileAssets.Select(c => c.ItemSpec))}");
+                    break;
+                }
+            }
+
+            if (!CompileAssets.Any())
+            {
+                Log.LogError($"Could not locate compile assets for any of the frameworks {String.Join(";", compileFxs.Select(fx => fx.ToString()))}");
+            }
+
+            foreach (var runtimeFx in runtimeFxs)
+            {
+                RuntimeAssets = _resolver.ResolveRuntimeAssets(runtimeFx, TargetRuntime)
+                                         .Where(ra => !NuGetAssetResolver.IsPlaceholder(ra))
+                                         .Select(ra => PackageItemAsResolvedAsset(_targetPathToPackageItem[ra]))
+                                         .ToArray();
+
+                if (RuntimeAssets.Any())
+                {
+                    Log.LogMessage($"Resolved runtime assets from {runtimeFx.ToString()}: {String.Join(";", RuntimeAssets.Select(r => r.ItemSpec))}");
+                    break;
+                }
+            }
+
+            if (!RuntimeAssets.Any())
```

Comment:
You missed this https://github.com/dotnet/buildtools/pull/770/files/99e2773de109d33c5a7a584ea33f2c9b5586390e#diff-73b43d708f5a6d42e26a2bfcf7eb1a08R83.  We default to use the compileFxs.  CompileFxs are required.